app/models/profile.rb

```diff
@@ -120,6 +128,22 @@ def has_proofed_before?
     Profile.where(user_id: user_id).where.not(activated_at: nil).where.not(id: self.id).exists?
   end
 
+  def irs_attempts_api_tracker
+    return @irs_attempts_api_tracker if defined?(@irs_attempts_api_tracker)
+    @irs_attempts_api_tracker = if initiating_service_provider&.irs_attempts_api_enabled?
+      IrsAttemptsApi::Tracker.new(
+        session_id: nil,
+        request: nil,
+        user: user,
+        sp: initiating_service_provider,
+        cookie_device_uuid: nil,
+        sp_request_uri: nil,
+        enabled_for_session: true,
+        analytics: nil, # FIXME, set this
```

Comment:
@zachmargolis You had very reasonably suggested I not keep this nil.

What I'm running into is that the Analytics code assumes session is not nil, with stuff like this failing if it is:

```ruby
  def update_session_events_and_paths_visited_for_analytics(event)
    @session[:events] ||= {}
    @session[:first_event] = !@session[:events].key?(event)
    @session[:events][event] = true
  end
```

The reference to `nil[:events]` raises an exception. It seems silly to construct a dummy session, but it also seems like making the analytics code support a nil session is pretty deep down a rabbit hole. And yet it _also_ seems like " 🤷 leave `analytics` nil, then, I guess" is less than ideal. I've got to call it a day here, but curious if you have an opinion on the most reasonable approach here.

Reply:
I would pass in a plain hash as "session" just to keep that codepath satisfied

```ruby
Analytics.new(
  # ...
  session: {},
  # ...
)
```